src/channel_internal.c

```diff
@@ -139,17 +151,29 @@ static void execute_callbacks(void* context)
         CHANNEL_OP* channel_op = (CHANNEL_OP*)context;
         CHANNEL_CALLBACK_RESULT result = channel_op->result; // local copy to make sure both callbacks are called with the same result
```

Comment:
>// local copy to make sure both callbacks are called with the same result [](http://example.com/codeflow?start=61&length=73)

This comment right here, what is the scenario where they are not called with the same result?

Reply:
cancel could get called between the execution of the callbacks